ext/mbstring/mb_gpc.c

```diff
@@ -235,23 +234,7 @@ const mbfl_encoding *_php_mb_encoding_handler_ex(const php_mb_encoding_handler_i
 	} else if (info->num_from_encodings == 1) {
 		from_encoding = info->from_encodings[0];
 	} else {
-		/* auto detect */
-		from_encoding = NULL;
-		identd = mbfl_encoding_detector_new(info->from_encodings, info->num_from_encodings, MBSTRG(strict_detection));
-		if (identd != NULL) {
-			n = 0;
-			while (n < num) {
-				mbfl_string string;
-				string.val = (unsigned char *)val_list[n];
-				string.len = len_list[n];
-				if (mbfl_encoding_detector_feed(identd, &string)) {
-					break;
-				}
-				n++;
-			}
-			from_encoding = mbfl_encoding_detector_judge(identd);
-			mbfl_encoding_detector_delete(identd);
-		}
+		from_encoding = mb_guess_encoding_for_strings((const unsigned char**)val_list, len_list, num, info->from_encodings, info->num_from_encodings, MBSTRG(strict_detection));
```

Comment:
Can't you move the const unsigned cast if you change
```c
val_list = (char **)ecalloc(num, sizeof(char *));
```
to

```c
val_list = (const unsigned char**)ecalloc(num, sizeof(char *));
```

and
```c
char *strtok_buf = NULL, **val_list = NULL;
```
to

```c
char *strtok_buf = NULL;
const unsigned char **val_list = NULL;
```

But then maybe leave those sort of fixes for a follow-up PR (or just ignore).

Reply:
Just tried that. It results in even more casts needed in other places.

Reply:
Yeah than leave it, I might have a go at this if I truly care :)